daml/ContingentClaims/FinancialClaim.daml

```diff
@@ -48,17 +48,24 @@ fixed principal coupon = floating (O.pure principal) (O.pure coupon)
 -- european (at $ date 2021 05 14) (Zero `Or` observe "SPX" - pure 4200)
 -- ```
 european : t -> Claim f t x a -> Claim f t x a -- TODO : consider swapping ordre of parameters for consistency with bermudan?
-european t u = When (at t) (u `Or` Zero)
+european maturity payoff = When (at maturity) (payoff `Or` Zero)
 
 -- | Bermudan option on the passed claim.
 bermudan : Claim f t x a -> [t] -> Claim f t x a
 bermudan u = apo \case
   (t :: ts) -> Left (european t u) `OrF` Right ts
   [] -> ZeroF
 
+-- | American option (knock-in). The lead parameter is the first possible acquisition date.
+american : Enum t => t -> t -> Claim f t x a -> Claim f t x a
+american t maturity payoff = 
+  Until (at (succ maturity)) $
+    Anytime ((>=) t) (payoff `Or` Zero)
+  where (>=) = at
```

Comment:
I don't get why you set `(>=)` to `at` here. `at` does date equality, so it reads now `anytime (today == t)`..?

Reply:
`at` has been changed; if you look at the definition it's `TimeGte` (time greater than or equal).

The thing is that when using it when `When`, requires you to take _the first instant_ when the value is true. So effectively, at behaves like equality.
`Anytime` differs from `When`, because it allows an acquisition region. So for `Anytime`, having an equality is not desireable; otherwise it's just the same as `When`.

I recognise this is confusing though. It was done to reduce the ADT and number of cases that we have to handle.